asm/cluster/istio-operator.yaml

```diff
@@ -15,37 +15,35 @@
 apiVersion: install.istio.io/v1alpha1
 kind: IstioOperator
 metadata:
-  clusterName: "PROJECT_ID/us-central1-c/asm-cluster" # {"type":"string","x-kustomize":{"partialSetters":[{"name":"gcloud.core.project","value":"PROJECT_ID"},{"name":"cluster-name","value":"asm-cluster"},{"name":"gcloud.compute.zone","value":"us-central1-c"}]}}
+  clusterName: "PROJECT_ID/us-central1-c/asm-cluster" # {"$ref":"#/definitions/io.k8s.cli.substitutions.cluster-name"}
 spec:
-  profile: asm-gcp
-  hub: gcr.io/gke-release/asm
-  tag: 1.6.4-asm.9
+  profile: asm-gcp # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.profile"}
+  hub: gcr.io/gke-release/asm # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.hub"}
+  tag: 1.6.5-asm.1 # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.tag"}
+  components:
+    pilot:
+      k8s:
+        env:
+        - name: SPIFFE_BUNDLE_ENDPOINTS
+          value: "PROJECT_ID.svc.id.goog|https://storage.googleapis.com/mesh-ca-resources/spiffe_bundle.json" # {"$ref":"#/definitions/io.k8s.cli.substitutions.spiffe-bundle-endpoints"}
+        resources:
+          requests:
+            cpu: 50m
```

Comment:
please carefully verified this 50m CPU will work for all scenarios and long-running ASD instances. 50m CPU is not tested in any other tests in OSS Istio and ASM.

Reply:
- Did performance tests on 3 x 2vCPU-node cluster via [fortio](https://github.com/fortio/fortio) with hundreds of concurrent connections and/or qps to onlineboutique application. 
- Managed to increase CPU consumption of onlineboutique pods [1], and triggered some errors with Envoy Proxy sidecar [2]. 
- Resource consumption for pilot stays low throughout the load testing (<=5m CPU, <30Mi Memory) 
- From [documentation](https://istio.io/latest/docs/ops/deployment/performance-and-scalability/#control-plane-performance) it looks like pilot CPU consumption scales with rate of deployment / configurations changes, and number of proxies. These numbers are expected to be very low for normal ASD deployments because 1) the tutorial encourages users to update deployment via ACM repo which is a manual process 2) the small size cluster won't be have enough CPU to schedule a lot more pods hence number of proxies will remain low.
- While there are no authoritative benchmark data for pilot with 50m requested CPU, there are other service mesh implementation e.g. [openshift  ](https://docs.openshift.com/container-platform/4.5/service_mesh/service_mesh_install/customizing-installation-ossm.html#ossm-cr-pilot_customizing-installation-ossm) with default cpu request of 10m. 

Is there any recommended testing specifically for pilot that I should additionally perform?

[1] Example `kubectl top` output when usage is at peak
```
NAME                                                  CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%   
gke-anthos-sample-cluste-default-pool-f2589fb0-dmp2   395m         20%    1471Mi          24%       
gke-anthos-sample-cluste-default-pool-f2589fb0-jn51   1209m        62%    1819Mi          30%       
gke-anthos-sample-cluste-default-pool-f2589fb0-xkc8   1565m        81%    1359Mi          22%       
NAMESPACE                  NAME                                                             CPU(cores)   MEMORY(bytes)   
config-management-system   git-importer-66657d7d9d-tcsj7                                    38m          40Mi            
config-management-system   monitor-7df4b4bdf-4md9k                                          1m           6Mi             
config-management-system   syncer-5f4b54bc45-892wj                                          5m           32Mi            
gatekeeper-system          gatekeeper-controller-manager-7b8f7ccf54-cjcts                   18m          29Mi            
gke-connect                gke-connect-agent-20200703-02-00-76544b8d46-bsqjd                5m           25Mi            
istio-system               istio-ingressgateway-5455c8ddb4-8jpkb                            43m          69Mi            
istio-system               istio-ingressgateway-5455c8ddb4-cgpwf                            26m          67Mi            
istio-system               istiod-55fc7c7f69-dtbpp                                          4m           32Mi            
istio-system               istiod-55fc7c7f69-sxgcq                                          4m           36Mi            
istio-system               promsd-6d7d8768f4-j7bq8                                          1m           24Mi            
kube-system                config-management-operator-6785f957f5-c58nd                      7m           19Mi            
kube-system                event-exporter-gke-6c9d8bd8d8-xkpb4                              1m           9Mi             
kube-system                fluentd-gke-8hlcs                                                593m         176Mi           
kube-system                fluentd-gke-fdh69                                                21m          167Mi           
kube-system                fluentd-gke-s99gj                                                896m         182Mi           
kube-system                fluentd-gke-scaler-cd4d654d7-f6rx4                               0m           39Mi            
kube-system                gke-metadata-server-7b7zt                                        1m           14Mi            
kube-system                gke-metadata-server-jsrlv                                        1m           13Mi            
kube-system                gke-metadata-server-zstss                                        2m           16Mi            
kube-system                gke-metrics-agent-88nf8                                          1m           22Mi            
kube-system                gke-metrics-agent-kdbvf                                          1m           22Mi            
kube-system                gke-metrics-agent-qjktg                                          1m           22Mi            
kube-system                kube-dns-56d8cd994f-q89rf                                        7m           33Mi            
kube-system                kube-dns-56d8cd994f-w7qxx                                        7m           33Mi            
kube-system                kube-dns-autoscaler-645f7d66cf-whs69                             1m           4Mi             
kube-system                kube-proxy-gke-anthos-sample-cluste-default-pool-f2589fb0-dmp2   2m           13Mi            
kube-system                kube-proxy-gke-anthos-sample-cluste-default-pool-f2589fb0-jn51   2m           14Mi            
kube-system                kube-proxy-gke-anthos-sample-cluste-default-pool-f2589fb0-xkc8   2m           13Mi            
kube-system                l7-default-backend-678889f899-2zlk6                              1m           2Mi             
kube-system                metrics-server-v0.3.6-7b7d6c7576-vqwxl                           3m           23Mi            
kube-system                netd-42smd                                                       1m           8Mi             
kube-system                netd-q9bwc                                                       0m           8Mi             
kube-system                netd-qs9wr                                                       1m           7Mi             
kube-system                prometheus-to-sd-5wxh6                                           1m           7Mi             
kube-system                prometheus-to-sd-qdvd9                                           1m           7Mi             
kube-system                prometheus-to-sd-qwwtz                                           0m           7Mi             
kube-system                stackdriver-metadata-agent-cluster-level-7f56c4f46d-4tbnh        7m           20Mi            
onlineboutique             adservice-6b9f9667fc-dwkrw                                       30m          185Mi           
onlineboutique             cartservice-7b4ffc8f7c-qtzhh                                     62m          107Mi           
onlineboutique             currencyservice-5b69f47d-rqllx                                   354m         187Mi           
onlineboutique             emailservice-797bd8568b-wxzrv                                    10m          86Mi            
onlineboutique             frontend-57d689bc7f-mz2js                                        456m         251Mi           
onlineboutique             loadgenerator-77f9869746-f79br                                   9m           74Mi            
onlineboutique             loadgenerator-77f9869746-qzl6g                                   9m           74Mi            
onlineboutique             paymentservice-75d69db68b-xjk5p                                  7m           82Mi            
onlineboutique             productcatalogservice-57bb7dc6f8-grpxx                           40m          75Mi            
onlineboutique             recommendationservice-747f574f54-f7mld                           19m          87Mi            
onlineboutique             redis-cart-5bc7f8777b-8z2h4                                      7m           41Mi            
onlineboutique             shippingservice-cdc6655dc-z6gcv                                  11m          58Mi            
onlineboutique             shoppingcartservice-768dd5479-98w97                              7m           59Mi            

```
[2] Errors from `istio-proxy` sidecar containers
```
2020-07-14T12:23:04.816682Z	info	transport: loopyWriter.run returning. connection error: desc = "transport is closing"
2020-07-14T12:26:15.656697Z	error	Request to probe app failed: Get "http://localhost:8080/_healthz": context deadline exceeded (Client.Timeout exceeded while awaiting headers), original URL path = /app-health/server/readyz
2020-07-14T12:27:04.853970Z	info	transport: loopyWriter.run returning. connection error: desc = "transport is closing"
```
[3] An ASM screenshot taken during load testing
![image](https://user-images.githubusercontent.com/2538581/87427692-c204e380-c613-11ea-8dee-ac32b9c762cf.png)

Reply:
Great work!